src/Wallabag/CoreBundle/Repository/EntryRepository.php

```diff
@@ -86,6 +86,36 @@ public function getBuilderForStarredByUser($userId)
     }
 
     /**
+     * Retrieves entries filtered with a search term for a user.
+     *
+     * @param int    $userId
+     * @param string $term
+     * @param strint $currentRoute
+     *
+     * @return QueryBuilder
+     */
+    public function getBuilderForSearchByUser($userId, $term, $currentRoute)
+    {
+        $qb = $this
+            ->getBuilderByUser($userId);
+
+        if ('starred' === $currentRoute) {
+            $qb->andWhere('e.isStarred = true');
+        } elseif ('unread' === $currentRoute) {
+            $qb->andWhere('e.isArchived = false');
+        } elseif ('archive' === $currentRoute) {
+            $qb->andWhere('e.isArchived = true');
+        }
+
+        $qb
+            ->andWhere('e.content LIKE :term OR e.title LIKE :term')->setParameter('term', '%'.$term.'%')
```

Comment:
Thank you @damienalexandre for your review 👍 

> to do "real" token based full text search directly with MySQL 

We also support Postgres and SQLite. I didn't find `MatchAgainst` in SQLite and Postgres folders in DoctrineExtensions.